pkg/sql/resolve_oid.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/errors"
+	"github.com/lib/pq/oid"
+)
+
+// ResolveOIDFromStringOrOID is part of tree.EvalPlanner.
+func (p *planner) ResolveOIDFromStringOrOID(
+	ctx context.Context, oidTyp *types.T, d tree.Datum,
```

Comment:
is it worth it to add another interface here that's only implemented by DOid and DString, so the type can be more restricted?

Reply:
See how this makes you feel.